### PR TITLE
[Enhancement] Support Prompt-to-prompt, ddim inversion and null-text inversion

### PR DESCRIPTION
**Pull request by FerryHuang**
Support Prompt-to-prompt, ddim inversion and null-text inversion.

Considering that these three methods share common features and can be combined, I implemented them as modules in this project.

## Checklist

**Before PR**:

- [√] I have read and followed the workflow indicated in the [CONTRIBUTING.md](https://github.com/open-mmlab/mmagic/blob/main/.github/CONTRIBUTING.md) to create this PR.
- [√] Pre-commit or linting tools indicated in [CONTRIBUTING.md](https://github.com/open-mmlab/mmagic/blob/main/.github/CONTRIBUTING.md) are used to fix the potential lint issues.
- [√] Bug fixes are covered by unit tests, the case that causes the bug should be added in the unit tests.
- [√] New functionalities are covered by complete unit tests. If not, please add more unit test to ensure the correctness.
- [√] The documentation has been modified accordingly, including docstring or example tutorials.

**After PR**:

- [ ] If the modification has potential influence on downstream or other related projects, this PR should be tested with some of those projects.
- [ ] CLA has been signed and all committers have signed the CLA in this PR.

### CODE REVIEW

File: projects/prompt_to_prompt/inversions/ddim_inversion.py [hunk 1]
```diff
@@ -0,0 +1,168 @@
...
+            beta_start=0.00085,
+            beta_end=0.012,
+            beta_schedule='scaled_linear',
+            clip_sample=False,
+            set_alpha_to_one=False)
+        self.model = model
+        self.num_ddim_steps = kwargs.pop('num_ddim_steps', 50)
+        self.guidance_scale = kwargs.pop('guidance_scale', 7.5)
+        self.tokenizer = self.model.tokenizer
+        self.model.schdueler = scheduler
```
Comment (LeoXing1996) [inline projects/prompt_to_prompt/inversions/ddim_inversion.py:40]: `schdueler` -> `scheduler`

**Review by LeoXing1996 — CHANGES_REQUESTED** [review]

File: projects/prompt_to_prompt/inversions/ddim_inversion.py [hunk 2]
```diff
@@ -0,0 +1,168 @@
...
+            else:
+                image = torch.from_numpy(image).float() / 127.5 - 1
+                image = image.permute(2, 0,
+                                      1).unsqueeze(0).to(self.model.device)
+                latents = self.model.vae.encode(image)['latent_dist'].mean
+                latents = latents * 0.18215
+        return latents
+
+    @torch.no_grad()
+    def init_prompt(self, prompt: str):
```
[the remark this answers is not in the document]
Comment (FerryHuang) [reply projects/prompt_to_prompt/inversions/ddim_inversion.py:125]: Thanks for your feedback and kind suggestion! This implementation intends to use text prompt as additional information to get the noise prediction so as to guide the inversion process to be more precise. I implement it with reference to the ddim inversion part of the original null-text inversion paper.
image
The _C_ here is likely to be the text prompt and it's actually the classifier-free guidance under the case of w=1, (the _pivot _pivot_ named by the author). So I vote for keeping it here for use in both inversions.

**Comment by LeoXing1996** [comment]
@FerryHuang Sorry for late response. The implementation is already good enough. However, there is one aspect that needs improvement: directly uploading images will increase the repository's size. Please change the images in the markdown to image hosting links. 

For example, you can drag and drop an image into the chat box in Github and a link will be generated. This link will be permanently saved in Github and you can use it in your markdown.

**Comment by FerryHuang** [comment]
> @FerryHuang Sorry for late response. The implementation is already good enough. However, there is one aspect that needs improvement: directly uploading images will increase the repository's size. Please change the images in the markdown to image hosting links.
> 
> For example, you can drag and drop an image into the chat box in Github and a link will be generated. This link will be permanently saved in Github and you can use it in your markdown.

Thanks for the trick! And changes have been made~

**Review by LeoXing1996 — APPROVED** [review]
LGTM! Great work!
